### PR TITLE
Whisker length, more precise description

### PR DESCRIPTION
**Pull request by RadostW**
See issue for discussion.
 https://github.com/matplotlib/matplotlib/issues/25134#issue-1568623320

## PR Summary
Slight change of the documentation wording to make it more precise.

## PR Checklist

**Documentation and Tests**

Not applicable: - [ ] Has pytest style unit tests (and `pytest` passes)

- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

Not applicable:  - [ ] New plotting related features are documented with examples.

### CODE REVIEW

**Review by tacaswell — APPROVED** [review]

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -3723,9 +3723,10 @@ def boxplot(self, x, notch=None, sym=None, vert=None, whis=None,
         Draw a box and whisker plot.
 
         The box extends from the first quartile (Q1) to the third
-        quartile (Q3) of the data, with a line at the median.  The
-        whiskers extend from the box by 1.5x the inter-quartile range
-        (IQR).  Flier points are those past the end of the whiskers.
+        quartile (Q3) of the data, with a line at the median. The
+        whiskers extend from the box to the farthest point within
+        1.5x the inter-quartile range (IQR) distance. Flier points
+        are those past the end of the whiskers.
```
Comment (jklymak) [inline lib/matplotlib/axes/_axes.py:3729]: Maybe?

```suggestion
        quartile (Q3) of the data, with a line at the median. The
        whiskers extend from the box to 1.5x the inter-quartile 
        range (IQR) distance from the median. Flier points
        are those past the end of the whiskers.
```

Comment (jklymak) [reply lib/matplotlib/axes/_axes.py:3729]: And actually I'm confused - is the ASCII art also incorrect?

Comment (RadostW) [reply lib/matplotlib/axes/_axes.py:3729]: No!
It's 1.5x IQR from the edge of the box, so `low = Q1 - 1.5 * IQR` location of the median has no influence here

Comment (jklymak) [inline lib/matplotlib/axes/_axes.py:3729]:
```suggestion
        quartile (Q3) of the data, with a line at the median. The
        whiskers extend from the box to 1.5x the inter-quartile 
        range (IQR) distance from the box. Flier points
        are those past the end of the whiskers.
```

Comment (jklymak) [reply lib/matplotlib/axes/_axes.py:3729]: OK, well, its still not clear then, but at least the ASCII art is correct...

Comment (RadostW) [reply lib/matplotlib/axes/_axes.py:3729]: Also the whole point of this change was to explain why sometimes whiskers are asymmetric (see https://github.com/matplotlib/matplotlib/issues/25134#issue-1568623320) and the proposal above https://github.com/matplotlib/matplotlib/pull/25135#discussion_r1095012398 doesn't resolve that.

Comment (jklymak) [reply lib/matplotlib/axes/_axes.py:3729]: also, is it the farthest data point and hence why you said "farthest point"?

Comment (RadostW) [reply lib/matplotlib/axes/_axes.py:3729]: Yes. 
For example if all points fit inside the [Q1 - Q3] range there would be no whiskers at all. An example of such data could be `vals = [1,1,1,3,3,3]`

In that sense the ASCII art is correct in spirit but does not address the weird cases.

Comment (jklymak) [reply lib/matplotlib/axes/_axes.py:3729]: So maybe just specify "data point" and distance from edge of box?

Comment (RadostW) [reply lib/matplotlib/axes/_axes.py:3729]: So:

The whiskers extend from the box to the farthest data point 
lying within 1.5x the inter-quartile range (IQR) from the box

?

Comment (jklymak) [reply lib/matplotlib/axes/_axes.py:3729]: Yes, that would make sense to me...

Comment (RadostW) [reply lib/matplotlib/axes/_axes.py]: There is a commit ready with the wording above

**Review by jklymak — APPROVED** [review]
Thanks for persevering explaining this to me!

**Comment by jklymak** [comment]
I took the liberty of squash merging.  Thanks a lot for fixing this and we hope to hear from you again...

**Pull request merged**